spec/defines/property_spec.rb

```diff
@@ -4,219 +4,270 @@
   let(:title) { 'some-property' }
   let(:directory) { '/tmp/mesos-conf' }
 
-  let(:params) {{
-    :value   => 'foo',
-    :dir     => directory,
-    :owner   => 'tester',
-    :group   => 'testers',
-  }}
-
-  it { should compile }
-
-  it 'should create a property file' do
-      should contain_file(
-        "#{directory}/#{title}"
-      ).with_content(/^foo$/).with({
-      'ensure'  => 'present',
-      'owner'   => 'tester',
-      'group'   => 'testers',
-      })
-  end
+  context 'with a string value' do
+    let(:params) do
+      {
+          :value => 'foo',
+          :dir => directory,
+      }
+    end
 
-  context 'with an empty string value' do
-    let(:params) {{
-      :value   => '',
-      :dir     => directory,
-    }}
+    it { is_expected.to compile.with_all_deps }
 
-    it 'should not contain a property file' do
-        should contain_file(
-          "#{directory}/#{title}"
-        ).with({
-        'ensure'  => 'absent',
-        })
+    it do
+      parameters = {
+          :ensure => 'present',
+          :content => "foo\n",
+      }
+      is_expected.to contain_file("#{directory}/#{title}").with(parameters)
     end
   end
 
-  context 'with an undef value' do
-    let(:params) {{
-      :value   => :undef,
-      :dir     => directory,
-    }}
+  context 'with an empty value' do
+    let(:params) do
+      {
+          :value => '',
```

Comment:
You're changing purpose of the test. The test was failing due to some other issue, see #70.